src/main/java/tree/ImpTargetTreeNode.kt

```diff
@@ -1,30 +1,89 @@
 package tree
 
-class ImpTargetTreeNode : TargetTreeNode {
-    override fun moveRight(): TargetTreeNode {
-        println("moveRight")
-        return this
+class ImpTargetTreeNode(
+    private val name: String = "",
+    private val tag: String = "",
+    private var id: Int = -1
+) : TargetTreeNode {
+
+    private var children: List<TargetTreeNode> = emptyList()
+    private var parent: TargetTreeNode? = null
+
+    override fun moveToRightSibling(): TargetTreeNode? {
+        val rightSibling = parent as? ImpTargetTreeNode
+        return rightSibling?.getRightSibling(this)
+    }
+
+    override fun moveToLeftSibling(): TargetTreeNode? {
+        if (parent != null)
+            return (parent!! as ImpTargetTreeNode).getLeftSibling(this)
+        return null
+    }
+
+    override fun moveToParent(): TargetTreeNode? {
+        return parent
+    }
+
+    override fun moveToFirstChild(): TargetTreeNode? {
+        if (children.isNotEmpty())
+            return children.first()
+        return null
+    }
+
+    override fun getName(): String {
+        return name
     }
 
-    override fun moveLeft(): TargetTreeNode {
-        println("moveLeft")
-        return this
+    override fun getTag(): String {
+        return tag
     }
 
-    override fun moveUp(): TargetTreeNode {
-        println("moveUp")
-        return this
+    override fun getId(): Int {
+        return id
     }
 
-    override fun moveDown(): TargetTreeNode {
-        println("moveDown")
-        return this
+    override fun setChildren(children: List<TargetTreeNode>) {
+        this.children = children
     }
 
-    override val name: String
-        get() = "name"
+    override fun getChildren(): List<TargetTreeNode> {
+        return children
+    }
+
+    override fun toString(): String {
+        var str = "$name:$tag($id)"
+        for (child in children) {
+            str += "(" + (child as ImpTargetTreeNode).toString() + ")"
```

Comment:
We have to options for string, make similar to input with string.prependIndent or () what we will choose?
@juanpa097 @DorianMoreno 
![](https://media.giphy.com/media/giEftvZuPAiZQBphEI/giphy.gif)

Reply:
Oh yes, I made this before we decided on making these tests similar to the input. So I'll try to do it that way.

Reply:
So be clear which way was chosen ?

Reply:
Print the tree with the same format in the test cases with indents

Reply:
I modified the function in order to print a tree similiar to the input one.